defaults/main.yml

```
@@ -33,7 +33,7 @@ kubectl_checksum_binary: "sha512:https://cdn.dl.k8s.io/release/v{{ kubectl_versi
 kubectl_bin_directory: "/usr/local/bin"
 
 # Directory to store the kubectl archive
-kubectl_tmp_directory: "{{ lookup('env', 'TMPDIR') | default('/tmp', true) }}"
+kubectl_tmp_directory: "{{ ansible_facts.env.TMPDIR | default('/tmp', true) }}"
```

```suggestion
kubectl_tmp_directory: "{{ ansible_env.TMPDIR | default('/tmp', true) }}"
```
Thanks for the PR! Officially this role only supports Linux and Debian 😉  . Nevertheless you're right `lookup('env', 'TMPDIR')` is evaluated on the controller node and not on the destination host. But `ansible_facts.env.TMPDIR` has also an issue IMHO. If `ansible_facts` aren't gathered (no `gather_facts: true` in the playbook), then `ansible_facts.env.TMPDIR` will be undefined. And I guess facts gathering is most probably not implemented everywhere. I mean at the end it's a variable and you can override its value 😉 Please try if `ansible_env.TMPDIR` works for you. I guess then the change would be acceptable.

---

Ah, I didn't even check if Mac was supported 😅 Yes for my case, I think it's totally just easiest if I just set a value (like `/tmp`) and not rely on the automatic detection. I also just checked if your suggestion works, and it does! I've updated the README snippet as well.

I'm not too familiar with Ansible, so I wasn't sure whether `ansible_facts.env` or `ansible_env` is better, but your point about the fact gathering makes sense to me!